core/libtexpdf-output.lua

```diff
@@ -109,9 +109,9 @@ SILE.outputters.libtexpdf = {
     return self:drawHbox(value, width)
   end,
 
-  _drawString = function(self, str, width)
+  _drawString = function(self, str, width, x_offset, y_offset)
     local x, y = self:getCursor()
-    pdf.setstring(x, y, str, string.len(str), self._font, width)
+    pdf.setstring(x+x_offset, y+y_offset, str, string.len(str), self._font, width)
```

Comment:
This will take care of the libtexpdf's internal comprehension of where the cursor is, but I think we need a call to self:setCursor() here to match our own internal comprehension of the same, otherwise they are going to be out of sync.

Reply:
Or maybe I'm still misunderstanding this. I'm surprised it passed the BBT-RTL test for rules.

Reply:
No! Offsets should not affect the cursor; that is the whole point of offsets.

Reply:
( https://simoncozens.github.io/fonts-and-layout//concepts.html#advance-and-positioning )

Reply:
Oh crikey. The rules test no longer mixes text strings & rules....

Reply:
The offsets should only non-zero when Harfbuzz gives us positioning information to draw a glyph at a certain displacement (e.g. mark positioning, cursive positioning). Any other time it should behave as before.